src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java

```diff
@@ -94,18 +94,18 @@ private void updateClaimStateJudicialReferral(
 
     private boolean isJudicialReferralAllowed(CaseData caseData) {
         return (caseData.isClaimantNotSettlePartAdmitClaim() || caseData.isFullDefence())
-            && (Objects.nonNull(caseData.getCaseDataLiP()) && caseData.getCaseDataLiP().hasClaimantNotAgreedToFreeMediation());
+            && (nonNull(caseData.getCaseDataLiP()) && caseData.getCaseDataLiP().hasClaimantNotAgreedToFreeMediation());
     }
 
     private void updateClaimEndState(AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response, CaseData updatedData) {
         if (updatedData.hasClaimantAgreedToFreeMediation() && updatedData.hasDefendantAgreedToFreeMediation()) {
             response.state(CaseState.IN_MEDIATION.name());
         } else if (updatedData.hasApplicant1SignedSettlementAgreement() && updatedData.hasApplicantAcceptedRepaymentPlan()) {
             response.state(CaseState.All_FINAL_ORDERS_ISSUED.name());
-        } else if (!updatedData.hasApplicantProceededWithClaim()) {
-            response.state(updatedData.isClaimantConfirmAmountPaidPartAdmit() || updatedData.hasDefendantPayedTheAmountClaimed()
-                               ? CaseState.CASE_SETTLED.name()
-                               : CaseState.CASE_DISMISSED.name());
+        } else if (!updatedData.hasApplicantProceededWithClaim() && nonNull(updatedData.getApplicant1ProceedWithClaim())) {
+            response.state(updatedData.hasDefendantPayedTheAmountClaimed() ? CaseState.CASE_SETTLED.name() : CaseState.CASE_DISMISSED.name());
+        } else if (updatedData.hasClaimantAcceptedAdmittedAmountPaid()) {
+            response.state(CaseState.CASE_SETTLED.name());
```

Comment:
Even claimant accepts the paid amount, has a chance to not settle the claim, I could this logic only checks the paid amount accepted but doesn't check the claim settle option chosen by claimant , so we shouldn't move to the claim settle state